### PR TITLE
proxy: Rely on hyperstart package

### PR DESCRIPTION
**Pull request by sboeuf**
This PR fixes issue #395. The code from proxy now relies on common hyperstart package instead of duplicating hyperstart implementations.

### CODE REVIEW

File: proxy/vm.go [hunk 1]
```diff
@@ -16,14 +16,12 @@ package main
...
 	"encoding/binary"
-	"errors"
 	"fmt"
-	"math"
 	"net"
 	"os"
 	"sync"
 
-	hyper "github.com/hyperhq/runv/hyperstart/api/json"
+	"github.com/sameo/virtcontainers/hyperstart"
```
Comment (jodh-intel) [inline proxy/vm.go]: Is this repo name going to change? If so, will github provide an auto-redirect?

Comment (sboeuf) [inline proxy/vm.go]: I don't know yet about that. @sameo has maybe an answer ?

**Comment by jodh-intel** [comment]
The Travis build is failing as it can't find the `virtcontainers` repo. A missing `go get`?

**Comment by sboeuf** [comment]
Yes it is failing because it does not find it. But I don't get it, travis is supposed to do a go get by itself, isn't it ?

**Comment by sboeuf** [comment]
I was assuming travis was doing a go get -u, but I updated the .travis.yml so that it gets our virtcontainers repo.

File: proxy/vm.go (hunk 1 above)
Comment (sameo) [inline proxy/vm.go]: @jodh-intel The repo name will change, and we will update cc-oci-runtime once it does. We won't close the `sameo/virtcontainers/` repo, we'll transition smoothly to the new one (either at `github.com/01org/` or `github.com/containers`.

**Comment by dlespiau** [comment]
I see that we're not vendoring virtcontainers. Is that the plan? no vendoring until some later point? or should we freeze virtcontainers and update it at key point?

**Comment by sameo** [comment]
@dlespiau FWIW we  just tagged virtcontainers v0.1.0. You may want to vendor that one.

**Comment by dlespiau** [comment]
Excellent, I think we'll vendor it so you are free to tweak the API without breaking the OCI runtime, we'll try to follow closely though to not accumulate technical debt..

**Comment by dlespiau** [comment]
So two things I can see (talking with @sboeuf  really) we may want to improve either in this PR on subsequent ones:

  - The proxy needs to split the I/O paquets and inspect the seq numbers to be able to route the paquets to the right shim. Right now that logic has been kept in the proxy, but I think it'd be (slightly) better to add an API to do that in the hyperstart package (you never know, may be even something you want to use to be able to tell stderr and stdout part)

  - The vendoring of virtcontainers.

I'll open issues with those, but for now:

lgtm

**Comment by jodh-intel** [comment]
lgtm

**Comment by chavafg** [comment]
qa-passed

**Comment by jodh-intel** [comment]
@sboeuf - hmm, same problem on this PR with pullapprove.

**Comment by jodh-intel** [comment]
This might point to a github -> pullapprove connectivity issue since this PR *has* been approved by Pullapprove: https://pullapprove.com/01org/cc-oci-runtime/pull-request/422/

**Comment by jodh-intel** [comment]
@sboeuf - please can you resolve the conflict and we'll see if pullapprove can now be coaxed into working once we re-ack.

**Comment by sboeuf** [comment]
@jodh-intel I have just solved the conflict. You can get this merge as soon as CI is OK :)

**Comment by dlespiau** [comment]
For the record, I've created https://github.com/01org/cc-oci-runtime/issues/438 and https://github.com/01org/cc-oci-runtime/issues/439 to track what was discussed above, unassigned at this point and P2 and P3 respectively.

**Comment by dlespiau** [comment]
lgtm

**Comment by jodh-intel** [comment]
lgtm

**Comment by chavafg** [comment]
qa-passed

File: proxy/vm.go (hunk 1 above)
Comment (gorozco1) [inline proxy/vm.go]: afaik we are building hyperstart from https://github.com/clearlinux/hyperstart, it wont be any conflicts?

Comment (sboeuf) [inline proxy/vm.go]: PR#422 allows the proxy to rely on the hyperstart package from virtcontainers. This means that we use the same codebase to discuss with hyperstart for both CC2.1 and CC3.0 projects. The link you mentioned " https://github.com/clearlinux/hyperstart" corresponds to the hyperstart binary which is running inside the VM itself.

**Pull request merged**